### ISSUE TITLE
Wide Open ipv6 Egress Being Created

### ISSUE THREAD

**Issue by lw-trectenwald** [terraform-aws-modules/terraform-aws-security-group#130]
Module: `v3.0.1` Terraform: `0.12.3`

---
I am creating a load balancer security group that does the following:

- __Allows HTTPS ingress from a list of CIDR ranges.__
- __Allows HTTPS egress to a specific security group.__

When I run the below, the ingress rules are fine. However, egress also creates a wide open IPv6 rule along with the proper one referencing a security group:

```
module "ext_alb_sg" {
  source              = "terraform-aws-modules/security-group/aws"
  name                = "prod-ph-ext-alb-sg"
  vpc_id              = data.terraform_remote_state.vpc.outputs.vpc.id
  ingress_rules       = ["https-443-tcp"]
  ingress_cidr_blocks = var.regional_accelerator_cidrs

  computed_egress_with_source_security_group_id = [
    {
      from_port                = 443
      to_port                  = 443
      protocol                 = "tcp"
      source_security_group_id = module.tomcat_sg.this_security_group_id
    }
  ]

  number_of_computed_egress_with_source_security_group_id = 1
}
```

The above creates egress like:

|Type|Protocol|Port Range|Destination|Description|
|----|----|----|----|----|
|HTTPS|TCP|443|::/0|Egress Rule|
|HTTPS|TCP|443|sg-1234567890|Egress Rule|

I'm not sure where the `::/0` wide open rule comes from. If I delete it, it'll recreate.

Thanks,
Tom

**Comment by lw-trectenwald** [comment]
I believe I may have found the issue. Line 525 of `main.tf` sets an IPv6 CIDR Block range that seems to default to `::/0` in the `aws_security_group_rule.computed_egress_with_source_security_group_id` resource. I don't think this should be there as the goal here is to make a rule with an SG not CIDR.

I cloned the repository, pointed at the module locally, commented out that line and tried again. All works fine now. I notice this defined at `aws_security_group_rule.ingress_with_source_security_group_id` as well so am thinking that may have the same issue. It looks like `egress_ipv6_cidr_blocks` defaults to `::/0`

```
variable "egress_ipv6_cidr_blocks" {
  description = "List of IPv6 CIDR ranges to use on all egress rules"
  type        = list(string)
  default     = ["::/0"]
}
```

I'm not sure it is possible to set a TF list to `null` (I tried but it still used the default), or empty `[]` ... gets an error. So, the variable defaults to wide open, then may be improperly called in the resources that involve source security groups.

Hope some of that helps.

**Comment by JamieGruener** [comment]
I just ran into this exact problem. The main reason I'm looking at this plugin is to create server/client Security Group pairs, but if all my egress gets full IPv6 access, that's kind of frustrating. It seems like a simple fix. Any workarounds available? (Other than pulling down the source and using it locally, I mean.)

**Comment by lw-trectenwald** [comment]
I was going to fork, modify and run the code locally.. but it just seemed a messy solution. I ended up using the built-in resources in TF. I define all ingress & egress rules via associated aws_security_group_rule resources (i.e. don't use the array approach in aws_security_group). That combined with some homegrown modules for repeatable patterns has worked just fine.

**Comment by ricardojdsilva87** [comment]
I'm having the same issue on the egress_with_source_security_group_id, the IPV6 range is always added with the needed SG on 2 separated lines

**Comment by antonbabenko** [comment]
This has been fixed in #148 and [v3.6.0](https://github.com/terraform-aws-modules/terraform-aws-security-group/releases/v3.6.0) has been just released.
